docs/design/2022-04-15-multi-schema-change.md

```diff
@@ -0,0 +1,137 @@
+
+On the other hand, if there is an error returned by any sub-job before all of them become non-revertible, the whole job is switched to a `rollingback` state. For the executed sub-jobs, we set them to `cancelling`; for the sub-job that have not been executed, we set them to `cancelled`.
+
+Finally, we consider the extreme case: an error occurs while all the sub-jobs are non-revertible. In this situation, we tend to believe that the error can be solved in a trivial way like retrying. This behavior is compatible with the current DDL implementation. Take `DROP COLUMN` as an example, once the column steps to the "Write-Only" state, there is no way to cancel this job.
+
+## Compatibility
+
+### Upgrade Compatibility
+
+When a TiDB cluster performs a rolling upgrade, there are 2 cases:
+
+1. DDL owner is the new version of TiDB. When users execute a Multi-Schema Change statement on an old version of TiDB, they receive an "Unsupported Multi-Schema Change" error message.
+2. DDL owner is the old version of TiDB. When users execute a Multi-Schema Change statement on a new version of TiDB, they receive the "invalid ddl job type" error message.
+
+In both cases, the Multi-Schema Change statement cannot be executed. Therefore, users should avoid executing Multi-Schema Change during a rolling upgrade.
+
+### MySQL Compatibility
+
+MySQL may reorder some schema changes:
+
+```SQL
+CREATE TABLE t (b INT, a INT, INDEX i(b));
+ALTER TABLE t DROP COLUMN b, RENAME COLUMN a TO b, ADD INDEX i(b), DROP INDEX i; -- success!
+
+SHOW CREATE TABLE t;
+```
+
+```console
++-------+-------------------------------------------------------------------------------------------------------------------------------+
+| Table | Create Table                                                                                                                  |
++-------+-------------------------------------------------------------------------------------------------------------------------------+
+| t     | CREATE TABLE `t` (
+ `b` int DEFAULT NULL,
+ KEY `i` (`b`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci |
++-------+-------------------------------------------------------------------------------------------------------------------------------+
+```
+
+MySQL's execution order is as follows:
+
+1. `DROP INDEX i`
+2. `DROP COLUMN b`
+3. `RENAME a TO b`
+4. `ADD INDEX i(b)`
+
+
+Since this behavior is a bit counter-intuitive, we decided not to be fully compatible with MySQL.
+
+```SQL
+ALTER TABLE t DROP COLUMN b, RENAME COLUMN a TO b, ADD INDEX i(b), DROP INDEX i;
+ERROR 1060 (42S21): Duplicate column name 'b'
+```
+
+TiDB validates the schema changes against a snapshot schema structure retrieved before the execution, regardless of the previous changes in the same DDL SQL statement.
+
+## Alternative Proposals
```

Comment:
Please also add the following alternatives:

* The 'job group' way.
* The existing way, which is a case-by-case implementation(for example, https://github.com/pingcap/tidb/pull/15540)